File: examples-sdk/zig/023-basic-hybrid-wasm-sidecar/main.zig
// 023-basic-hybrid-wasm-sidecar — Zig SDK equivalent
// Compile: vil compile --from zig --input 023-basic-hybrid-wasm-sidecar/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("hybrid-pipeline", 8080);
    var pipeline = vil.Service.init("pipeline");
    pipeline.endpoint("GET", "/", "index");
    pipeline.endpoint("POST", "/validate", "validate_order");
    pipeline.endpoint("POST", "/price", "calculate_price");
    pipeline.endpoint("POST", "/fraud", "fraud_check");
    pipeline.endpoint("POST", "/order", "process_order");
    server.service(&pipeline);
    server.compile();
}

File: examples-sdk/go/023-basic-hybrid-wasm-sidecar/main.go
// 023-basic-hybrid-wasm-sidecar — Go SDK equivalent
// Compile: vil compile --from go --input 023-basic-hybrid-wasm-sidecar/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("hybrid-pipeline", 8080)

	pipeline := vil.NewService("pipeline")
	pipeline.Endpoint("GET", "/", "index")
	pipeline.Endpoint("POST", "/validate", "validate_order")
	pipeline.Endpoint("POST", "/price", "calculate_price")
	pipeline.Endpoint("POST", "/fraud", "fraud_check")
	pipeline.Endpoint("POST", "/order", "process_order")
	s.Service(pipeline)

	s.Compile()
}
